### PR TITLE
Dev to master 

### PR DESCRIPTION
**Pull request by JL-GH**

### CODE REVIEW

File: stripeServer/index.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const express = require('express');
+
```
Comment (HypeDis) [inline stripeServer/index.js]: nice!

File: server/db/user_models/user.js [hunk 2]
```diff
@@ -14,40 +14,84 @@ const Users = db.define('users', {
     primaryKey: true
   },
   firstName: {
-    type: STRING
+    type: STRING,
+    allowNull: false,
```
Comment (HypeDis) [inline server/db/user_models/user.js]: why wasnt this here from the beginnning?

File: server/api/products.js [hunk 3]
```diff
@@ -23,7 +21,7 @@ router.get('/', (req, res, next) => {
     });
 });
 router.get('/:id', (req, res, next) => {
-  const id = parseInt(req.params.id);
+  // const id = parseInt(req.params.id);
```
Comment (HypeDis) [inline server/api/products.js]: you should validate that 'id' is in the right format.
so if id should be a number you should check that it is in fact a number before making the query

File: bash-X.X.tar.gz [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
```
Comment (HypeDis) [inline bash-X.X.tar.gz]: wat dis?
delete it or ignore it

File: app/redux/authentication.js [hunk 5]
```diff
@@ -104,7 +104,58 @@ export const initialLogInAttempt = () => {
...
+
+        // now deal with cart items
+        // want to put them in local storage
+
+        // check if there are existing items in local storage so we can merge if needed
+        if (localStorage.getItem(localStorageKey)) {
+          const localStorageItems = JSON.parse(
+            localStorage.getItem(localStorageKey)
+          );
+          // add the items to the redux store
```
Comment (HypeDis) [inline app/redux/authentication.js]: delete commented code

File: app/components/root.jsx [hunk 6]
```diff
@@ -44,7 +45,9 @@ export default class Root extends Component {
             <Route path="/myaccount" component={MyAccount} />
             <Route path="/checkout" component={CheckoutGuestAddress} exact />
             <Route path="/checkout2" component={CheckoutGuestPay} exact />
-            <Route path="/success" component={Success} exact />
+            <Route path="/success/:id" component={Success} exact />
+            <Route path="/addproduct" component={AddProduct} exact />
+            <Route path="/editprofile" component={EditUserProfile} exact />
```
Comment (HypeDis) [inline app/components/root.jsx]: i want to see a 404 route at the end of the switch

File: app/components/editUserProfile.jsx [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+      error: false,
+      errorText: ''
+    };
+  }
+
+  componentDidMount() {
+    const { activeUser } = this.props;
+    console.log('activeUser', activeUser);
+    this.setState({
+      firstName: activeUser.firstName,
```
Comment (HypeDis) [inline app/components/editUserProfile.jsx]: could probably just do this.setState(activUser)

Comment (HypeDis) [reply app/components/editUserProfile.jsx]: if this is coming from redux state then just use the redux state and dispatches
action creator: changeValue({key: value})
reducer: return {...state, ...changedValue}

File: app/components/editUserProfile.jsx [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+      shippingCity,
+      shippingState,
+      shippingZip,
+      billingAddress1,
+      billingAddress2,
+      billingCity,
+      billingState,
+      billingZip
+    } = this.state;
+    if (firstName === '') {
```
Comment (HypeDis) [inline app/components/editUserProfile.jsx]: create a function called validateInputs(inputs)

File: app/components/Success.jsx [hunk 9]
```diff
@@ -1,6 +1,5 @@
 import React from 'react';
```
Comment (HypeDis) [inline app/components/Success.jsx]: make filename more descriptive

File: app/components/ProductCatTile.jsx [hunk 10]
```diff
@@ -4,12 +4,8 @@ import { getProductThunk } from '../redux/singleProduct.js';
 import { makeStyles } from '@material-ui/core/styles';
```
Comment (HypeDis) [inline app/components/ProductCatTile.jsx]: fix filename

File: app/components/CheckoutGuestPay.jsx [hunk 11]
```diff
@@ -105,6 +121,8 @@ class GuestPayment extends Component {
               id="billingCity"
               name="billingCity"
               label="City"
+              error={this.state.cityErr}
+              helperText={this.state.cityHelper}
               value={this.state.billingCity}
               onChange={this.handleChange}
             />
```
Comment (HypeDis) [inline app/components/CheckoutGuestPay.jsx]: create  a generateError(this.state.inputError){
if(inputError){
  return <div>my error</div>
}
}

Comment (HypeDis) [reply app/components/CheckoutGuestPay.jsx]: disregard

File: app/components/AddProduct.jsx [hunk 12]
```diff
@@ -0,0 +1,289 @@
...
+import TextField from '@material-ui/core/TextField';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+import { FormHelperText } from '@material-ui/core';
+import Button from '@material-ui/core/Button';
+import { createProduct } from '../redux/products';
+const sizes = ['one size', 'XS', 'S', 'M', 'L', 'XL'];
+const gender = ['F', 'M', 'N'];
+
+const categories = [
```
Comment (HypeDis) [inline app/components/AddProduct.jsx]: you could probably create a separate file with these types and impor them

File: app/components/root.jsx (hunk 6 above)
Comment (hertzj) [reply app/components/root.jsx]: will be in once 'DRY Text Fields and 404 page' PR is accepted

File: server/db/user_models/user.js (hunk 2 above)
Comment (hertzj) [reply server/db/user_models/user.js]: we removed some validation to make manual testing faster (I'm ready to be yelled at for this :) )

**Pull request merged**